src/curvefit/solvers/solvers.py

```diff
@@ -14,12 +15,52 @@ class SolverNotDefinedError(Exception):
     pass
 
 
+@dataclass(frozen=True)
+class SolverOptions:
+    """
+    {begin_markdown SolverOptions}
+
+    # `curvefit.solvers.solvers.SolverOptions`
+    ## Parameter specification for a Solver
+
+    TODO: make docs after we decide what parameters we want to be here. It would be perfect to have
+    some universal set of parameters like (numiter, rtol, atol) and then translate it to a solver-specific
+    parameters via .to_dict() but I am not sure if it's always possible.
+
+    ## Arguments
+
+    {end_markdown SolverOptions}
+    """
+    ftol: float
+    gtol: float
+    maxiter: int
+    disp: bool
+
+    def __post_init__(self):
+        assert self.maxiter > 0, "maxiter should be positive"
+
+    def to_dict(self, solver_type="sciopt"):
+        # Different solvers have different naming for their options:
```

Comment:
FYI, dataclass has a `asdict()` function.

Reply:
That's the thing: translating parameters to dict is a solver-specific operation. Suppose you have SolverOptions, and you want to pass them to two solvers: SciOpt and IPOpt. Just casting asdict(options) won't be sufficient because they expect different parameters.

Reply:
see slack conversation